fury/ui.py

```diff
@@ -4325,6 +4326,278 @@ def directory_click_callback(self, i_ren, _obj, listboxitem):
+
+        self.drop_down_button = Button2D(
+            icon_fnames=self._icon_files, size=self.drop_button_size)
+
+        self.drop_down_menu = ListBox2D(
+            values=self.items, multiselection=False,
+            font_size=self.font_size, line_spacing=self.line_spacing,
+            text_color=self.menu_txt_color, selected_color=self.selected_color,
+            unselected_color=self.unselected_color,
+            scroll_bar_active_color=self.scroll_active_color,
+            scroll_bar_inactive_color=self.scroll_inactive_color,
+            background_opacity=self.menu_opacity,
+            reverse_scrolling=self.reverse_scrolling, size=self.drop_menu_size)
+
+        self.drop_down_menu.set_visibility(False)
+
+        self.panel = Panel2D(self.panel_size, opacity=0.0)
+
+        self.panel.add_element(self.selection_box, (0.001, 0.7))
+        self.panel.add_element(self.drop_down_button, (0.8, 0.7))
+        self.panel.add_element(self.drop_down_menu, (0, 0))
+
+        if self.draggable:
+            self.drop_down_button.on_left_mouse_button_dragged =\
+                self.left_button_dragged
+            self.drop_down_menu.panel.background.on_left_mouse_button_dragged\
+                = self.left_button_dragged
+            self.selection_box.on_left_mouse_button_dragged =\
+                self.left_button_dragged
+            self.selection_box.background.on_left_mouse_button_dragged =\
+                self.left_button_dragged
+
+            self.drop_down_button.on_left_mouse_button_pressed =\
+                self.left_button_pressed
+            self.drop_down_menu.panel.background.on_left_mouse_button_pressed\
+                = self.left_button_pressed
+            self.selection_box.on_left_mouse_button_pressed =\
+                self.left_button_pressed
+            self.selection_box.background.on_left_mouse_button_pressed =\
+                self.left_button_pressed
+        else:
+            self.panel.background.on_left_mouse_button_dragged =\
+                lambda i_ren, _obj, _comp: i_ren.force_render
+            self.drop_down_menu.panel.background.on_left_mouse_button_dragged\
+                = lambda i_ren, _obj, _comp: i_ren.force_render
+
+        # Handle mouse wheel events on the slots.
+        for slot in self.drop_down_menu.slots:
+            slot.add_callback(
+                slot.textblock.actor, "LeftButtonPressEvent",
+                self.select_option_callback)
+
+            slot.add_callback(
+                slot.background.actor, "LeftButtonPressEvent",
+                self.select_option_callback)
+
+            self.drop_down_button.on_left_mouse_button_clicked = \
+                self.menu_toggle_callback
+
+        # Offer some standard hooks to the user.
+        self.on_change = lambda ui: None
+
+    def _get_actors(self):
+        """ Get the actors composing this UI component.
+        """
+        return self.panel.actors
+
+    def resize(self, size):
```

Comment:
Can you add a small docstring?  Thanks !

Reply:
Hello @skoudoro , I have added the changes.

Reply:
Thank you @Nibba2018 ! When the CI's finished, I will merge it and check your tutorial PR.